crates/languages/src/python.rs

```diff
@@ -732,6 +748,31 @@ async fn get_worktree_venv_declaration(worktree_root: &Path) -> Option<String> {
     Some(venv_name.trim().to_string())
 }
 
+fn get_venv_parent_dir(env: &PythonEnvironment) -> Option<PathBuf> {
+    // If global, we aren't a virtual environment
+    if let Some(kind) = env.kind
+        && is_python_env_global(&kind)
+    {
+        return None;
+    }
+
+    let venv = env
+        .executable
+        .as_ref()
+        .and_then(|p| p.parent())
+        .and_then(|p| p.parent());
+
+    // Check to be sure we are a virtual environment using pet's most generic
+    // virtual environment type, VirtualEnv
+    if let Some(venv) = venv
+        && is_virtualenv_dir(venv)
+    {
+        venv.parent().map(|parent| parent.to_path_buf())
+    } else {
+        None
+    }
```

Comment:
```suggestion
    // Check to be sure we are a virtual environment using pet's most generic
    // virtual environment type, VirtualEnv
    let venv = env
        .executable
        .as_ref()
        .and_then(|p| p.parent())
        .and_then(|p| p.parent())
        .filter(is_virtualenv_dir)?;

    venv.parent().map(|parent| parent.to_path_buf())
```

Reply:
I'm getting a type mismatch on the filter when I do it this way, `&&Path` instead of `&Path`. I'll do it with a closure instead?

Reply:
Yep, sure. I wrote this patch directly within GitHub interface, so there may be problems like such.